src/NumericalAlgorithms/LinearOperators/MeanValue.hpp

```diff
@@ -22,17 +22,18 @@ class Mesh;
 
 /*!
  * \ingroup NumericalAlgorithmsGroup
- * \brief Compute the mean value of a grid-function over a manifold.
- * \f$mean value = \int f dV / \int dV\f$
+ * \brief Compute the mean value of a grid function over a manifold.
  *
- * \remarks The mean value is computed on the reference element(s).
- * \note The mean w.r.t. a different set of coordinates x can be computed
- * by pre-multiplying the argument f by the Jacobian J = dx/dxi of the mapping
- * from the reference coordinates xi to the coordinates x.
+ * Given a function \f$f\f$, compute its mean value \f$\bar{f}\f$ with respect
+ * to the reference coordinates \f$(\xi, \eta, \zeta)\f$. In 1 dimension, e.g.,
+ * \f$\bar{f} = \int_{-1}^1 f d\xi \Big/ \int_{-1}^1 d\xi\f$.
  *
- * \returns the mean value of `f` on the manifold
- * \param f the grid function of which to find the mean.
- * \param mesh the Mesh of the manifold on which f is located.
+ * The mean w.r.t. a different set of coordinates \f$x\f$ can be computed by
```

Comment:
This is not true.  You need to modify the integral in the divisor as well which cannot be done using this function...

Reply:
I agree. What do you think of the following:

- edit this text to reflect that this function only works in logical coords.
- add a new commit (probably in a new PR) that adds a new function `mean_value(data, mesh, jacobian)` that does the divisor appropriately for the coordinates associated with the jacobian.